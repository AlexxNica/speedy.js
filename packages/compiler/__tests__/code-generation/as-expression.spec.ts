import {runCases} from "./code-generation-test-case-runner";

runCases("AsExpression", "as-expression");
