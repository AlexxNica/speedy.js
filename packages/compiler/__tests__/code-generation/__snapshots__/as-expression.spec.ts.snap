// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsExpression as-int 1`] = `
"; ModuleID = 'as-expression/as-int.ts'
source_filename = \\"as-expression/as-int.ts\\"
target datalayout = \\"e-m:e-p:32:32-i64:64-n32:64-S128\\"
target triple = \\"wasm32-unknown-unknown\\"

%class.Math = type { { [5 x i8]* }* }

@Array_name = private unnamed_addr constant [6 x i8] c\\"Array\\\\00\\"
@Array_type_descriptor = private constant { [6 x i8]* } { [6 x i8]* @Array_name }
@Math_name = private unnamed_addr constant [5 x i8] c\\"Math\\\\00\\"
@Math_type_descriptor = private constant { [5 x i8]* } { [5 x i8]* @Math_name }
@Math_object = private constant %class.Math { { [5 x i8]* }* @Math_type_descriptor }
@Math_ptr = private constant %class.Math* @Math_object

define void @_asInt() {
entry:
  %intValue = alloca i32, align 4
  %AsInt32 = call i32 @toInt32d(double 3.212000e+00)
  store i32 %AsInt32, i32* %intValue, align 4
  ret void
}

; Function Attrs: alwaysinline readnone
declare i32 @toInt32d(double) #0

attributes #0 = { alwaysinline readnone }
"
`;

exports[`AsExpression as-number 1`] = `
"; ModuleID = 'as-expression/as-number.ts'
source_filename = \\"as-expression/as-number.ts\\"
target datalayout = \\"e-m:e-p:32:32-i64:64-n32:64-S128\\"
target triple = \\"wasm32-unknown-unknown\\"

%class.Math = type { { [5 x i8]* }* }

@Array_name = private unnamed_addr constant [6 x i8] c\\"Array\\\\00\\"
@Array_type_descriptor = private constant { [6 x i8]* } { [6 x i8]* @Array_name }
@Math_name = private unnamed_addr constant [5 x i8] c\\"Math\\\\00\\"
@Math_type_descriptor = private constant { [5 x i8]* } { [5 x i8]* @Math_name }
@Math_object = private constant %class.Math { { [5 x i8]* }* @Math_type_descriptor }
@Math_ptr = private constant %class.Math* @Math_object

define void @_asNumber() {
entry:
  %numberValue = alloca double, align 8
  store double 3.000000e+00, double* %numberValue, align 8
  ret void
}
"
`;
